src/adapt/layout.js

```diff
@@ -204,6 +204,63 @@ adapt.layout.BreakPosition.prototype.findAcceptableBreak = function(column, pena
 adapt.layout.BreakPosition.prototype.getMinBreakPenalty = function() {};
 
 /**
+ * @return {number} next penalty for this break position
+ */
+adapt.layout.BreakPosition.prototype.getNextMinBreakPenalty = function() {};
+
+/**
+ * @return {vivliostyle.layoututil.RepetitiveElements}
+ */
+adapt.layout.BreakPosition.prototype.getRepetitiveElements = function() {};
+
+/**
+ * @abstract
+ * @constructor
+ * @implements {adapt.layout.BreakPosition}
+ */
+adapt.layout.AbstractBreakPosition = function() {};
+
+/**
+ * @abstract
+ */
+adapt.layout.AbstractBreakPosition.prototype.findAcceptableBreak = function(column, penalty) {};
+
+/**
+ * @abstract
+ */
+adapt.layout.AbstractBreakPosition.prototype.getMinBreakPenalty = function() {};
+
+/**
+ * @override
+ */
+adapt.layout.AbstractBreakPosition.prototype.getNextMinBreakPenalty = function() {
+    var repetitiveElements =  this.getRepetitiveElements();
+    return this.getMinBreakPenalty()
+        + (repetitiveElements ? repetitiveElements.getNextPenaltyIncreasement() : 0);
+};
+
+/**
+ * @abstract
+ */
+adapt.layout.AbstractBreakPosition.prototype.getRepetitiveElements = function() {};
+
+/**
+ * @param {adapt.vtree.NodeContext} nodeContext
+ * @return {vivliostyle.layoututil.RepetitiveElements}
+ */
+adapt.layout.AbstractBreakPosition.prototype.retrieveRepetitiveElements = function(nodeContext) {
+    if (!nodeContext) return null;
+    var formattingContext = nodeContext.formattingContext;
+    while (formattingContext) {
+        var repetitiveElements = formattingContext.getRepetitiveElements();
+        if (repetitiveElements) return repetitiveElements;
+        formattingContext = formattingContext.getParent();
+    }
+    return null;
+};
```

Comment:
I think we need to take into account multiple `RepetitiveElements` when the fragmented element has multiple ancestors each of which has its own `RepetitiveElements`.
I don't think this issue should be addressed in this PR, but we might need to deal with it later.